### PR TITLE
Change master in ruby exercises links to main

### LINKED ISSUE #23491: Update outdated links to ruby-exercises repo
After changing from `master` to `main`, some of the links to the `ruby-exercises` repos are now outdated (but still functional w/ a notification of the change). 

So far, I've noticed it in:

* Assignment 3 Basic Data Types: https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-programming/lessons/basic-data-types
  - `ruby_programming/basic_ruby/basic_data_types.md`
* Assignment 3 Arrays: https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-programming/lessons/arrays
  - `ruby_programming/basic_ruby/arrays.md`
* Assignment 2 Hashes: https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-programming/lessons/hashes
  - `ruby_programming/basic_ruby/hashes.md`
* Assignment 4 Methods: https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-programming/lessons/methods
  - `ruby_programming/basic_ruby/methods.md`
* Assignment 4 Basic Enumerable Methods: https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-programming/lessons/basic-enumerable-methods
  - `ruby_programming/basic_ruby/basic_enumerable_methods.md`
* Assignment 2 Predicate Enumerable Methods: https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-programming/lessons/predicate-enumerable-methods
  - `ruby_programming/basic_ruby/predicate_enumerable_methods.md`
  
  I may have missed some.
  I will volunteer if it sits for a while, but I would like to give someone the opportunity to contribute.
  Briefly discussed with @thatblindgeye in the Discord (where he suggested I make a new issue).

### PR DESCRIPTION
**Pull request by drossano**
- [X] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).
 - [X] The title summarizes the change and where it happened, for example: "Fixes punctuation in Clean Code lesson".
 - [X] If the PR is related to an open issue, use a [relevant keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) and reference it with the `#` sign and the issue number.
 - [X] You have previewed your markdown (if any) in the [Odin Markdown Preview Tool](https://www.theodinproject.com/lessons/preview)
 - [X] If changes are requested, please make the changes in a timely manner. After you submit the changes, request another review from the maintainer (top right).

#### 1. Describe the changes made and include why they are necessary or important:
Changed all instances of 'master' to 'main' in all links to Ruby exercises in the basic ruby lessons. 

#### 2. Related Issue

Closes #23491

### CODE REVIEW

**Review by CouchofTomato — APPROVED** [review]

**Review by CouchofTomato — CHANGES_REQUESTED** [review]
There is a merge conflict that needs to be resolved

File: ruby_programming/basic_ruby/arrays.md [hunk 1]
```diff
@@ -141,8 +141,8 @@ Here is a brief look at some other common array methods you might run into:
...
 2. Read [The Definitive Guide to Ruby Arrays](https://www.rubyguides.com/2015/05/ruby-arrays/) by Jesus Castello.
-3. Complete the [array](https://github.com/TheOdinProject/ruby-exercises/tree/master/ruby_basics) exercises from the [ruby-exercises repo](https://github.com/TheOdinProject/ruby-exercises) that you previously cloned. 
-4. Let's get some practice reading documentation. You'll feel just like a real programmer! What do you think the methods `#clear`, `#insert`, `#sample`, `#shuffle`, and `#uniq` do? Look these methods up in the [Array class documentation](https://ruby-doc.org/core-3.0.3/Array.html). Were you close?
+3. Complete the [array](https://github.com/TheOdinProject/ruby-exercises/tree/main/ruby_basics) exercises from the [ruby-exercises repo](https://github.com/TheOdinProject/ruby-exercises) that you previously cloned. 
+4. Let's get some practice reading documentation. You'll feel just like a real programmer! What do you think the methods `#clear`, `#insert`, `#sample`, `#shuffle`, and `#uniq` do? Look these methods up in the [Array class documentation](https://ruby-doc.org/core-2.7.1/Array.html). Were you close?
```
Comment (zachmmeyer) [inline ruby_programming/basic_ruby/arrays.md:145]: Can we switch to using the latest docs for Array here?

```suggestion
4. Let's get some practice reading documentation. You'll feel just like a real programmer! What do you think the methods `#clear`, `#insert`, `#sample`, `#shuffle`, and `#uniq` do? Look these methods up in the [Array class documentation](https://ruby-doc.org/core/Array.html). Were you close?
```

**Review by zachmmeyer — CHANGES_REQUESTED** [review]
I have a single request in regards to linking to the docs here.

File: ruby_programming/basic_ruby/arrays.md (hunk 1 above)
Comment (CouchofTomato) [reply ruby_programming/basic_ruby/arrays.md]: @zachmmeyer Do we want to link to 3.1 docs when we tell students to use 3.0?

**Review by zachmmeyer — APPROVED** [review]
Thank you!

**Pull request merged**